src/web/controllers/index.js

```diff
@@ -369,31 +369,46 @@ router.get('/avatars', function (req, res) {
     }
 });
 
-router.get('/rankings', function (req, res) {//"Please login first"
+router.get('/rankings', function (req, res) {
     res.setHeader('Content-Type', 'application/json');
     var data = [];
+
+    // Debugging: Log session data
+    console.log("Session data:", req.session);
+
     if (req.query.type === 'friends') {
         if (req.session) {
             if (req.session.account_id) {
                 var game_id = req.session.game_id;
                 var rank = req.session.rank;
                 var acc_id = req.session.account_id;
+
+                // Debugging: Log account ID
+                console.log("Account ID:", acc_id);
+
                 req.db.connection.getConnection().then(conn => {
-                    conn.query('SELECT u.IdAcc, u.gp, u.game_id, u.rank FROM users u INNER JOIN friends ds ON u.IdAcc = ds.id_amigo INNER JOIN accounts a ON u.IdAcc = a.Id ANd ds.id_yo = ? ORDER BY u.gp DESC', [acc_id])
+                    conn.query('SELECT u.IdAcc, u.gp, u.game_id, u.rank FROM users u INNER JOIN friends ds ON u.IdAcc = ds.id_amigo INNER JOIN accounts a ON u.IdAcc = a.Id AND ds.id_yo = ? ORDER BY u.gp DESC', [acc_id])
```

Comment:
handling nested structures only generates callbackhells and is complicated to handle. Please improve this